core/src/main/scala/org/apache/spark/memory/DefaultMemoryManager.scala

```diff
@@ -0,0 +1,116 @@
+class DefaultMemoryManager extends StoreUnifiedManager with Logging {
+
+  private val memoryForObject: ObjectLongHashMap[(String, MemoryMode)]
+  = ObjectLongHashMap.withExpectedSize[(String, MemoryMode)](16)
+
+
+  private val managerId = "DefaultMemoryManager"
+
+  override def acquireStorageMemoryForObject(objectName: String,
+      blockId: BlockId,
+      numBytes: Long,
+      memoryMode: MemoryMode,
+      buffer: UMMMemoryTracker,
+      shouldEvict: Boolean): Boolean = {
+    logDebug(s"Acquiring DefaultManager memory for $objectName $numBytes")
+    if (SparkEnv.get ne null) {
+      val success = SparkEnv.get.memoryManager.acquireStorageMemory(blockId, numBytes, memoryMode)
+      memoryForObject.addTo(objectName -> memoryMode, numBytes)
+      success
+    } else {
+      true
+    }
+  }
+
+  // This is not be called in connector mode. This api is called from
+  // region.clear hence only applicable for local mode
+  override def dropStorageMemoryForObject(
+      objectName: String,
+      memoryMode: MemoryMode,
+      ignoreNumBytes: Long): Long = {
+    if (SparkEnv.get ne null) {
+      val key = objectName -> memoryMode
+      val bytesToBeFreed = memoryForObject.getLong(key)
+      val numBytes = Math.max(0, bytesToBeFreed - ignoreNumBytes)
+      logDebug(s"Dropping $managerId memory for $objectName =" +
+          s" $numBytes (registered=$bytesToBeFreed)")
+      if (numBytes > 0) {
+        SparkEnv.get.memoryManager.releaseStorageMemory(numBytes, memoryMode)
+        memoryForObject.removeAsLong(key)
+      }
+    }
+    0L
+  }
+
+  override def releaseStorageMemoryForObject(
+      objectName: String,
+      numBytes: Long,
+      memoryMode: MemoryMode): Unit = {
+    logDebug(s"Releasing DefaultManager memory for $objectName $numBytes")
+    if (SparkEnv.get ne null) {
+      SparkEnv.get.memoryManager.releaseStorageMemory(numBytes, memoryMode)
+      val key = objectName -> memoryMode
+      if (memoryForObject.containsKey(key)) {
+        if (memoryForObject.addTo(key, -numBytes) == numBytes) {
+          memoryForObject.removeAsLong(key)
+        }
+      }
+    }
+  }
+
+  override def getStoragePoolMemoryUsed(memoryMode: MemoryMode): Long = 0L
+
+  override def getStoragePoolSize(memoryMode: MemoryMode): Long = 0L
+
+  override def getExecutionPoolUsedMemory(memoryMode: MemoryMode): Long = 0L
+
+  override def getExecutionPoolSize(memoryMode: MemoryMode): Long = 0L
+
+  override def getOffHeapMemory(objectName: String): Long = 0L
+
+  override def hasOffHeap: Boolean = false
+
+  override def logStats(): Unit = logInfo("No stats for NoOpSnappyMemoryManager")
+
+  override def changeOffHeapOwnerToStorage(buffer: ByteBuffer,
+      allowNonAllocator: Boolean): Unit = {}
+
+  override def shouldStopRecovery(): Boolean = false
+
+  override def initMemoryStats(stats: MemoryManagerStats): Unit = {}
+
+  override def close(): Unit = {}
+
+  /**
+    * Clears the internal map
+    */
+  override def clear(): Unit = {}
```

Comment:
Have you verified that none of these empty methods are invoked from the store layer? Perhaps they don't need to be in interface at all or else at least some of them should return proper values reading from SparkEnv's memoryManager (like offheap related ones).

Reply:
The empty methods means they are only called from store layer, and makes sense in Snappy UMM. As default will be used in environment where store is not there, hence empty.

Reply:
Isn't SNAP-2084 for the local[*] mode where SnappyUMM may not have been explicitly configured? For that case callbacks from store will happen. For example whether off-heap has been configured or not which will effect the BufferAllocator.

Reply:
You mean to say if a local mode application configures offheap. Ok. In such scenarios we might need these APIs.